edx_solutions_api_integration/courses/utils.py

```diff
@@ -0,0 +1,107 @@
+from completion_aggregator.models import Aggregator
+from django.db.models import Q, Sum
+
+
+def _get_filtered_aggregation_queryset(course_key, exclude_users, org_ids=None, group_ids=None):
+    queryset = Aggregator.objects.filter(
+        course_key__exact=course_key,
+        user__is_active=True,
+        user__courseenrollment__is_active=True,
+        user__courseenrollment__course_id__exact=course_key,
+        aggregation_name='course',
+    ).exclude(user_id__in=exclude_users)
```

Comment:
This code seems too tightly coupled to the `Aggregator` internals. Can we make this call a Python or REST API from `completion_aggregator` rather than using its models directly?

Reply:
Agreed.  

But also, a lot of this filtering seems excessively specific for the aggregator api.  It would be more straightforward, and more broadly useful (across completion, grades, etc.) to just retrieve all the objects, and then perform further filtering in apros.  Specifically, I don't know that the API needs to care about whether the enrollment is active, or about the provided list of "excluded_users."  Or even probably if the user is_active or not.